### PR TITLE
feat(ec2): add `enableAcceptanceRequired` function for `VpcEndpointService`

### LINKED ISSUE #30192: aws_ec2.VpcEndpointService: Modify endpoint acceptance setting
### Describe the feature

When creating an Endpoint service using aws_ec2.VpcEndpointService, the acceptance_required field can be set. But, there is no method to update this flag after the endpoint service has been created. It is possible to modify this setting in console as well as the lower level construct aws_ec2.CfnVPCEndpointService. ``

### Use Case

Adding this feature in aws_ec2.VpcEndpointService will bring it to the level of console and lower level construct. With this feature, I can disable it in CDK for a short time to add intended connections and enable it again for security. Currently, someone with admin privileges need to monitor the console for requests, accept it manually and then ket the deployment continue which adds unnecessary overhead to automating the process.

### Proposed Solution

Add a method to toggle the flag acceptance_required for Endpoint services created using the construct aws_ec2.VpcEndpointService

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.88.0

### Environment details (OS name and version, etc.)

MacOS 14.3

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #30192.

### Reason for this change

Adding this feature in aws_ec2.VpcEndpointService will bring it to the level of console and lower level construct. With this feature, I can disable it in CDK for a short time to add intended connections and enable it again for security. Currently, someone with admin privileges need to monitor the console for requests, accept it manually and then ket the deployment continue which adds unnecessary overhead to automating the process.

### Description of changes

Add `enableAcceptanceRequired` method to the `ec2.VpcEndpointService` class.

### Description of how you validated changes

Add both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ec2/README.md [hunk 1]
```diff
@@ -1121,6 +1121,19 @@ Note: The domain name must be owned (registered through Route53) by the account
 The VpcEndpointServiceDomainName will handle the AWS side of domain verification, the process for which can be found
 [here](https://docs.aws.amazon.com/vpc/latest/userguide/endpoint-services-dns-validation.html)
 
+You can set `acceptanceRequired` not only `acceptanceRequired` prop but also `enableAcceptanceRequired()` method.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ec2/README.md:1124]:
```suggestion
To update the `acceptanceRequired` property after initialization use the `enableAcceptanceRequired()` method:
```

File: packages/aws-cdk-lib/aws-ec2/README.md [hunk 2]
```diff
@@ -1121,6 +1121,19 @@ Note: The domain name must be owned (registered through Route53) by the account
...
+
+```ts
+declare const networkLoadBalancer: elbv2.NetworkLoadBalancer;
+
+const endpointService = new ec2.VpcEndpointService(this, 'EndpointService', {
+  vpcEndpointServiceLoadBalancers: [networkLoadBalancer],
+});
+
+// update acceptanceRequired
+endpointService.enableAcceptanceRequired(false);
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-ec2/README.md:1134]:
```suggestion

const endpointService = new ec2.VpcEndpointService(this, 'EndpointService', {
  vpcEndpointServiceLoadBalancers: [networkLoadBalancer],
  acceptanceRequired: true,
});

endpointService.enableAcceptanceRequired(false);
```

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍 Left a couple of suggestions on the documentation

**Comment by badmintoncryer** [comment]
@lpizzinidev Thank you for your review!

**Comment by Leo10Gama** [comment]
Closing this PR as, unfortunately, we cannot accept changes like this. By allowing this value to be altered later in execution, it becomes unclear what the source of truth is for this property. There isn't a good enough reason to change the value outside of the constructor, and while adding in this functionality would cover a few specific use cases of toggling it off and on to add connections, it would ultimately degrade the user experience overall.

**Pull request closed without merging**